RTEMS build broken

**Describe the bug**
Needs updates to fix issues related to `OS_Milli2Ticks`
- Missing prototype
- References to old function prototype still exist in some RTEMS source files.

**To Reproduce**
Build default configuration for RTEMS 4.11.  Fails to compile with errors such as:
```
/home/joe/code/cfecfs/github/osal/src/os/rtems/src/os-impl-countsem.c: In function 'OS_CountSemTimedWait_Impl':
/home/joe/code/cfecfs/github/osal/src/os/rtems/src/os-impl-countsem.c:212:5: error: implicit declaration of function 'OS_Milli2Ticks' [-Werror=implicit-function-declaration]
     TimeInTicks = OS_Milli2Ticks(msecs);
     ^
```

**Expected behavior**
Should compile cleanly.

**System observed on:**
Ubuntu 20.04 (host) cross buiding for RTEMS 4.11.3 / pc-686 BSP

**Additional context**
Typo noted in #614 but there are more problems than just that....

**Reporter Info**
Joseph Hickey, Vantage Systems, Inc.

--- a/src/os/rtems/src/os-impl-binsem.c
+++ b/src/os/rtems/src/os-impl-binsem.c
@@ -39,6 +39,7 @@
 #include "os-impl-binsem.h"
 #include "os-shared-binsem.h"
 #include "os-shared-idmap.h"
+#include "os-shared-timebase.h"
 
 
 /****************************************************************************************
@@ -246,7 +247,7 @@ int32 OS_BinSemTimedWait_Impl (uint32 sem_id, uint32 msecs)
     rtems_status_code status;
     int               TimeInTicks;
 
-    if (OS_Milli2Ticks(msecs, &TimInTicks) != OS_SUCCESS)
+    if (OS_Milli2Ticks(msecs, &TimeInTicks) != OS_SUCCESS)
     {
         return OS_ERROR;
     }

--- a/src/os/rtems/src/os-impl-countsem.c
+++ b/src/os/rtems/src/os-impl-countsem.c
@@ -40,6 +40,7 @@
 
 #include "os-shared-countsem.h"
 #include "os-shared-idmap.h"
+#include "os-shared-timebase.h"
 
 /****************************************************************************************
                                      DEFINES
@@ -207,9 +208,12 @@ int32 OS_CountSemTake_Impl (uint32 sem_id)
 int32 OS_CountSemTimedWait_Impl (uint32 sem_id, uint32 msecs)
 {
     rtems_status_code status;
-    uint32            TimeInTicks;
+    int               TimeInTicks;
 
-    TimeInTicks = OS_Milli2Ticks(msecs);
+    if (OS_Milli2Ticks(msecs, &TimeInTicks) != OS_SUCCESS)
+    {
+        return OS_ERROR;
+    }
 
     status = rtems_semaphore_obtain(OS_impl_count_sem_table[sem_id].id, RTEMS_WAIT, TimeInTicks);
     if (status == RTEMS_TIMEOUT)

--- a/src/os/rtems/src/os-impl-queues.c
+++ b/src/os/rtems/src/os-impl-queues.c
@@ -40,6 +40,8 @@
 
 #include "os-shared-queue.h"
 #include "os-shared-idmap.h"
+#include "os-shared-timebase.h"
+
 
 /****************************************************************************************
                                      DEFINES
@@ -162,6 +164,7 @@ int32 OS_QueueGet_Impl (uint32 queue_id, void *data, uint32 size, uint32 *size_c
     int32              return_code;
     rtems_status_code  status;
     rtems_interval     ticks;
+    int                tick_count;
     rtems_option       option_set;
     size_t             rtems_size;
     rtems_id           rtems_queue_id;
@@ -182,8 +185,14 @@ int32 OS_QueueGet_Impl (uint32 queue_id, void *data, uint32 size, uint32 *size_c
     else
     {
         option_set = RTEMS_WAIT;
+
         /* msecs rounded to the closest system tick count */
-        ticks = OS_Milli2Ticks(timeout);
+        if (OS_Milli2Ticks(timeout, &tick_count) != OS_SUCCESS)
+        {
+            return OS_ERROR;
+        }
+
+        ticks = (rtems_interval)tick_count;
     }
 
     /*

--- a/src/os/rtems/src/os-impl-tasks.c
+++ b/src/os/rtems/src/os-impl-tasks.c
@@ -40,6 +40,7 @@
 
 #include "os-shared-task.h"
 #include "os-shared-idmap.h"
+#include "os-shared-timebase.h"
 
 /****************************************************************************************
                                      DEFINES
@@ -199,15 +200,28 @@ void OS_TaskExit_Impl()
  *-----------------------------------------------------------------*/
 int32 OS_TaskDelay_Impl (uint32 milli_second)
 {
-    rtems_interval     ticks;
+    int     tick_count;
+    int32   return_code;
 
-    ticks = OS_Milli2Ticks(milli_second);
+    return_code = OS_Milli2Ticks(milli_second, &tick_count);
 
-    rtems_task_wake_after(ticks);
+    if (return_code != OS_SUCCESS)
+    {
+        /*
+         * always want to do some form of delay, because if
+         * this function becomes a no-op then this might create a
+         * tight loop that doesn't ever yield the CPU - effectively
+         * locking the system in an RTOS environment.
+         */
+        tick_count = 10;
+    }
+
     /*
     ** Always successful ( from RTEMS docs )
     */
-    return (OS_SUCCESS);
+    rtems_task_wake_after((rtems_interval)tick_count);
+
+    return (return_code);
 
 } /* end OS_TaskDelay_Impl */
 

